pkg/scheduler/framework/interface.go

```diff
@@ -187,13 +188,19 @@ type PluginToStatus map[string]*Status
 
 // Merge merges the statuses in the map into one. The resulting status code have the following
 // precedence: Error, UnschedulableAndUnresolvable, Unschedulable.
-func (p PluginToStatus) Merge() *Status {
+// In addition to the merged status, it also returns a string set contains failed plugin names.
+func (p PluginToStatus) Merge() (*Status, sets.String) {
```

Comment:
If we add plugin name to status, then I would change merge such that the plugin name is the one that determined the code (line 207 below).

Reply:
One thing to keep in mind: in the scheduler, we generally only have one status, because we break at the first failure.
It's the cluster autoscaler that cares about running all plugins. However, the configuration is available in the policy API. It's not available in component config.

For the purpose of generality, shouldn't we keep all plugins, knowing that generally we only have one?

Reply:
> we generally only have one status, because we break at the first failure

True for non-Filter plugins. For Filter plugins, if no feasible node is found, we record the failures on each node, and return to generate an aggregated error.

Reply:
But in that case we don't use Merge.
Merge is used for failures on the same Node.

Reply:
We did use:

https://github.com/kubernetes/kubernetes/blob/44f8fcfe9482ecd6da7e9e88e0e25387ff2ef7fd/pkg/scheduler/framework/runtime/framework.go#L644-L645

Reply:
Those are results for one node.

Reply:
You're right, I misread. Then actually this method only makes sense when `runAllFilters` is true.